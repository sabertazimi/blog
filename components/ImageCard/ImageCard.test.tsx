import { render } from '@utils'
import ImageCard from './ImageCard'

describe('ImageCard', () => {
  test('should render placeholder image correctly (snapshot)', () => {
    const { container } = render(<ImageCard />)

    expect(container).toMatchSnapshot()
  })

  test('should render local image correctly (snapshot)', () => {
    const { container } = render(<ImageCard src="/public/images/landing.jpg" />)

    expect(container).toMatchSnapshot()
  })
})
