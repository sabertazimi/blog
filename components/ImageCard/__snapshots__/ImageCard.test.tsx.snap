// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard should render local image correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered ant-card-small text-center card"
  >
    <div
      class="ant-card-cover"
    >
      <div
        class="ant-image"
      >
        <img
          class="ant-image-img"
          src="/public/images/landing.jpg"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta justify-center"
      />
    </div>
  </div>
</div>
`;

exports[`ImageCard should render placeholder image correctly (snapshot) 1`] = `
<div>
  <div
    class="ant-image"
    title="Broken Image"
  >
    <img
      alt="Broken Image"
      class="ant-image-img"
      src="error"
    />
    <div
      class="ant-image-mask"
    >
      <div
        class="ant-image-mask-info"
      >
        <span
          aria-label="eye"
          class="anticon anticon-eye"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="eye"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
            />
          </svg>
        </span>
        Preview
      </div>
    </div>
  </div>
</div>
`;
